File: backend/app/wechat/message_parser.py
# -*- coding: utf-8 -*-
"""将 iLink 消息转换为 Agent 可处理的文本消息。"""
from __future__ import annotations

from typing import Any

from .models import IncomingText


def parse_text_messages(payload: dict[str, Any]) -> list[IncomingText]:
    messages = []
    for message in payload.get('msgs') or []:
        if message.get('message_type') == 2:
            continue
        text_parts = []
        for item in message.get('item_list') or []:
            if item.get('type') == 1 and (item.get('text_item') or {}).get('text'):
                text_parts.append(str(item['text_item']['text']))
        text = ''.join(text_parts).strip()
        if not text:
            continue
        messages.append(IncomingText(
            message_id=str(message.get('message_id') or message.get('seq') or ''),
            from_user_id=str(message.get('from_user_id') or ''),
            to_user_id=str(message.get('to_user_id') or ''),
            context_token=str(message.get('context_token') or ''),
            text=text,
        ))
    return messages

File: server/src/wechat/messageParser.ts
import type { IncomingText } from './models.js';

export function parseTextMessages(payload: Record<string, unknown>): IncomingText[] {
  const messages: IncomingText[] = [];
  for (const raw of Array.isArray(payload.msgs) ? payload.msgs : []) {
    const message = raw as Record<string, unknown>;
    if (message.message_type === 2) continue;
    const textParts: string[] = [];
    for (const rawItem of Array.isArray(message.item_list) ? message.item_list : []) {
      const item = rawItem as Record<string, unknown>;
      if (item.type === 1) {
        const textItem = item.text_item as Record<string, unknown> | undefined;
        if (textItem && textItem.text) textParts.push(String(textItem.text));
      }
    }
    const text = textParts.join('').trim();
    if (!text) continue;
    messages.push({
      message_id: String(message.message_id || message.seq || ''),
      from_user_id: String(message.from_user_id || ''),
      to_user_id: String(message.to_user_id || ''),
      context_token: String(message.context_token || ''),
      text,
    });
  }
  return messages;
}
